provider/k8s/client.go

```diff
@@ -1,296 +1,274 @@
 package k8s
 
 import (
-	"crypto/tls"
-	"crypto/x509"
-	"encoding/json"
-	"fmt"
-	"github.com/containous/traefik/log"
-	"github.com/parnurzeal/gorequest"
-	"net/http"
-	"net/url"
-	"strings"
-)
+	"time"
 
-const (
-	// APIEndpoint defines the base path for kubernetes API resources.
-	APIEndpoint        = "/api/v1"
-	extentionsEndpoint = "/apis/extensions/v1beta1"
-	defaultIngress     = "/ingresses"
-	namespaces         = "/namespaces/"
+	"k8s.io/client-go/1.5/kubernetes"
+	"k8s.io/client-go/1.5/pkg/api"
+	"k8s.io/client-go/1.5/pkg/api/v1"
+	"k8s.io/client-go/1.5/pkg/apis/extensions/v1beta1"
+	"k8s.io/client-go/1.5/pkg/fields"
+	"k8s.io/client-go/1.5/pkg/labels"
+	"k8s.io/client-go/1.5/pkg/runtime"
+	"k8s.io/client-go/1.5/pkg/watch"
+	"k8s.io/client-go/1.5/rest"
+	"k8s.io/client-go/1.5/tools/cache"
 )
 
+const resyncPeriod = time.Minute * 3
```

Comment:
this looks good. We could go for 5 minutes to keep parity with what kubedns does https://github.com/kubernetes/kubernetes/blob/ff8e780c30884161700ce741743455f1afc2a02f/pkg/dns/dns.go#L55